support-frontend/assets/helpers/existingPaymentMethods/existingPaymentMethods.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+// ----- Imports ----- //
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+import { fetchJson } from 'helpers/fetch';
+import { logException } from '../logger';
+import { DirectDebit, ExistingCard, ExistingDirectDebit, Stripe } from '../paymentMethods';
+import { getPaymentLabel } from '../checkouts';
+
+// ----- Types ----- //
+
+export type ExistingPaymentMethodSubscription = {
+  isActive: boolean;
+  isCancelled: boolean,
+  name: string
+}
+
+type ExistingPaymentType = 'Card' | 'DirectDebit';
+
+export type PartialExistingPaymentMethod = {
+  paymentType: ExistingPaymentType;
+};
+
+export type FullDetailExistingPaymentMethod = {
+  billingAccountId: string,
+  subscriptions: ExistingPaymentMethodSubscription[],
+  card?: string,
+  mandate?: string,
+  // can't use Intersection types to avoid repeating this as it breaks isFullDetailExistingPaymentMethod because; Flow!
+  paymentType: ExistingPaymentType
+};
+
+export type ExistingPaymentMethod = PartialExistingPaymentMethod | FullDetailExistingPaymentMethod;
+
+// ----- Functions ----- //
+
+function isFullDetailExistingPaymentMethod(existingPaymentMethod: ExistingPaymentMethod): boolean %checks {
+  return !!existingPaymentMethod.billingAccountId;
+}
+
+function sendGetExistingPaymentMethodsRequest(
+  isoCurrency: IsoCurrency,
+  storeResponse: ExistingPaymentMethod[] => void,
+): void {
+  fetchJson(
+    `${window.guardian.existingPaymentOptionsEndpoint}?currencyFilter=${isoCurrency}`,
```

Comment:
could we abstract this into `helpers/routes`? I'm not super happy with using `window.guardian` liberally as it's a magic global and it makes doing SSR and other optimizations very hard